### PR TITLE
Fix logfile file descriptor server leak

### PR DESCRIPTION
**Pull request by crazyfermions**
* Version of iperf3 (or development branch, such as `master` or
  `3.1-STABLE`) to which this pull request applies: master

* Issues fixed (if any): #1360 

* Brief description of code changes (suitable for use as a commit message):

```
Fix logfile fd leak in server

Makes server clean-up more robust, prevent fd leak and closing reopened fds by improving handing of fds (ensure -1 is set to an fd if it is closed to prevent closing an unrelated file if fd is reused). 
```

### CODE REVIEW

**Review by bmah888 — CHANGES_REQUESTED** [review]
A couple places where we need curly braces to make the conditionals work right when setting CPU affinity or doing JSON output.

**Comment by bmah888** [comment]
Hey @crazyfermions I think I like the basis of your PR better than mine (#1368) because you blow away the variables that keep track of sockets after the sockets get closed. That's better than what I was doing. There are a couple of minor fix-ups needed, but I think we should try to get #1369 into shape and use that to fix bug #1360. I haven't had a chance to run-test this code yet.

**Comment by bmah888** [comment]
Oh and I forgot to say thanks for submitting this PR!

File: src/iperf_server_api.c [hunk 1]
```diff
@@ -451,10 +461,12 @@ iperf_run_server(struct iperf_test *test)
 
     if (test->affinity != -1)
 	if (iperf_setaffinity(test, test->affinity) != 0)
+            cleanup_server(test);
 	    return -2;
```
[the remark this answers is not in the document]
Comment (crazyfermions) [reply src/iperf_server_api.c]: Of course, you are absolutely right.

**Comment by crazyfermions** [comment]
I fixed the embarrassing bug you caught. Do you plan on stashing this pull request? Otherwise, I thing it would make sense to rebase it and merge the fix with the other commit to reduce noise.

**Comment by bmah888** [comment]
Thanks for the changes. Lemme test / play around with your code a bit. If all goes well, I'm thinking to maybe squash and merge all of the changes in this PR as a single commit onto `master`. (In theory no further action on your part would be needed.)

**Review by bmah888 — APPROVED** [review]

**Comment by bmah888** [comment]
Thanks again for the fix!

**Pull request merged**